### ISSUE TITLE
How do I create a new yaml document

### ISSUE THREAD

**Issue by ws-way** [TomWright/dasel#37]
Hello @TomWright ,

Small question, but I am unable to figure it out, how do I create this yaml output?

```
metadata:
  annotations:
     node.longhorn.io/default-disks-config: '[ { "name":"fast",  "path":"/mnt/data-fast1", "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1", "allowScheduling":true, "tags":["slow"]} ]'
  labels:
    node.longhorn.io/create-default-disk: config
```

Thanks!

**Comment by TomWright** [comment]
Are you referring to the JSON content within `node.longhorn.io/default-disks-config`?

On another note, please use 3 backticks for code highlighting instead of 1. That way new lines etc are displayed properly.

**Comment by ws-way** [comment]
Thanks, I would to create the complete data structure from scratch

**Comment by TomWright** [comment]
You can use `dasel put string` with a blank input to create new documents.

E.g.

```
echo '' | dasel put string -p yaml some.path.here value
some:
  path:
    here: value
```

You can then pipe that output into another dasel command:
```
echo '' | dasel put string -p yaml 'some.path.here' value | dasel put string -p yaml 'some.path.too' value2
some:
  path:
    here: value
    too: value2
```

I just made a release that will allow you to use `.` in the property name by escaping it. Download dasel `v1.5.0` and then the following should work:

```
echo '' | dasel put string -p yaml 'metadata.annotations.node\.longhorn\.io\/default-disks-config' '[ { "name":"fast",  "path":"/mnt/data-fast1", "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1", "allowScheduling":true, "tags":["slow"]} ]' | dasel put string -p yaml 'metadata.annotations.node\.longhorn\.io\/create-default-disk' config
```

If you want to save this file just add `> myfile.yaml` at the end of the command.

I've just ran this locally with `v1.5.0`:
```
echo '' | dasel put string -p yaml 'metadata.annotations.node\.longhorn\.io\/default-disks-config' '[ { "name":"fast",  "path":"/mnt/data-fast1", "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1", "allowScheduling":true, "tags":["slow"]} ]' | dasel put string -p yaml 'metadata.annotations.node\.longhorn\.io\/create-default-disk' config
metadata:
  annotations:
    node.longhorn.io/create-default-disk: config
    node.longhorn.io/default-disks-config: '[ { "name":"fast",  "path":"/mnt/data-fast1",
      "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1",
      "allowScheduling":true, "tags":["slow"]} ]'
```

**Comment by ws-way** [comment]
Hi @TomWright,

Thanks, looking good. One question, this part:

```
            "    node.longhorn.io/default-disks-config: '[ { \"name\":\"fast\",  \"path\":\"/mnt/data-fast1\", \"allowScheduling\":true, \"tags\":[\"fast\"]}, { \"name\":\"slow\",  \"path\":\"/mnt/data-slow1\", \"allowScheduling\":true, \"tags\":[\"slow\"]} ]'" \
```

Is it possible to create this as an array? As far as I understand it, this part is an array and I do need 1 to x entries in this part, depending on the number of disks.

Thanks!

**Comment by TomWright** [comment]
Just append `[]` to the selector you want to be an array.

**Comment by ws-way** [comment]
Hi @TomWright ,

Thanks, I really appreciate the help you are giving, could you give me an example how to create this with an array?
 
```
metadata:
  annotations:
    node.longhorn.io/create-default-disk: config
    node.longhorn.io/default-disks-config: '[ { "name":"fast",  "path":"/mnt/data-fast1",
      "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1",
      "allowScheduling":true, "tags":["slow"]} ]'
```

The part of the disks-config is what I am looking for.

Thanks again

**Comment by TomWright** [comment]
Those values are in a JSON string. Do you want them as JSON or as YAML?

**Comment by ws-way** [comment]
Hi @TomWright ,

That should be JSON, thanks.

**Comment by TomWright** [comment]
```
echo '' | dasel put string -p yaml 'metadata.annotations.node\.longhorn\.io\/default-disks-config.[]' '[ { "name":"fast",  "path":"/mnt/data-fast1", "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1", "allowScheduling":true, "tags":["slow"]} ]' | dasel put string -p yaml 'metadata.annotations.node\.longhorn\.io\/create-default-disk' config
```

**Comment by ws-way** [comment]
Thanks!

**Comment by ws-way** [comment]
Hi @TomWright ,

Could you please take a look at this again?

I am trying to use your example, but it is different from my example. In your example config (last line) is also on metadata.annotation, but in my question it is on metadata.labels. When I replace the word annotation with labels it is not working, also not when I am splitting it into 2 commands.

This is not giving the line with the labels:
```
echo '' | dasel put string -p yaml 'metadata.annotations.node\.longhorn\.io\/default-disks-config' '[ { "name":"fast",  "path":"/mnt/data-fast1", "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1", "allowScheduling":true, "tags":["slow"]} ]' | dasel put string -p yaml 'metadata.labels.node\.longhorn\.io\/create-default-disk' config
```

Also splitting it into:

```
longhorn_disk_annotation=$(echo '' | dasel put string -p yaml 'metadata.annotations.node\.longhorn\.io\/default-disks-config' '[ { "name":"fast",  "path":"/mnt/data-fast1", "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1", "allowScheduling":true, "tags":["slow"]} ]')
result=$(echo $longhorn_disk_annotation | dasel put string -p yaml 'metadata.labels.node\.longhorn\.io\/create-default-disk' config)
```
How can I get this working?

**Comment by TomWright** [comment]
What is the final output you want to see?

**Comment by ws-way** [comment]
This :-)

```
metadata:
  annotations:
    node.longhorn.io/default-disks-config: '[ { "name":"fast",  "path":"/mnt/data-fast1", "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1", "allowScheduling":true, "tags":["slow"]} ]'
  labels:
    node.longhorn.io/create-default-disk: config
```

**Comment by ws-way** [comment]
Hello @TomWright,

I have tried it with variables and with files, the second put string is not working. Is this a bug or am I doing something wrong?

**Comment by TomWright** [comment]
I'll take a look at this today and get back to you

**Comment by ws-way** [comment]
Great, thanks!

**Comment by TomWright** [comment]
Hey @ws-way, this looks like a bug.

I'm debugging it now and will release a fix ASAP. Thanks for your patience 👍

**Comment by ws-way** [comment]
Ok, thanks, good luck with the fix

**Comment by TomWright** [comment]
OK the issue is fixed as of `v1.5.1`.

```
echo '' | dasel put string -p yaml 'metadata.annotations.node\.longhorn\.io\/default-disks-config' '[ { "name":"fast",  "path":"/mnt/data-fast1", "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1", "allowScheduling":true, "tags":["slow"]} ]' | dasel put string -p yaml 'metadata.labels.node\.longhorn\.io\/create-default-disk' 'config'
```

Output is:
```
metadata:
  annotations:
    node.longhorn.io/default-disks-config: '[ { "name":"fast",  "path":"/mnt/data-fast1",
      "allowScheduling":true, "tags":["fast"]}, { "name":"slow",  "path":"/mnt/data-slow1",
      "allowScheduling":true, "tags":["slow"]} ]'
  labels:
    node.longhorn.io/create-default-disk: config
```

**Comment by ws-way** [comment]
Looks it is working, thanks again!
